teps/0013-graceful-sidecar-termination.md

```diff
@@ -0,0 +1,226 @@
+---
+title: sidecar-termination-control-flag
+authors:
+  - "@shahnewazrifat"
+creation-date: 2020-08-31
+last-updated: 2020-08-31
+status: in pr review
+---
+
+# TEP-0013: Enable graceful sidecar termination/Give termination control to the task author.
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+- [Design Details](#design-details)
+- [Advantages](#advantages)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Currently, the sidecar feature in a tekton taskrun is not behaving like a typical kubernetes sidecar should. 
+The sidecar gets terminated immediately when the taskrun fails or completes without giving the sidecar enough time to 
+do what it originally intended to do (for example, upload logs, process artifacts etc.)
+In some cases the sidecar should be able to choose when to terminate itself. A sidecar is just a kubernetes container. 
+It's able to terminate itself with the command completion.
+
+
+This proposal is introducing a flag `waitForSidecar` which is a per-sidecar configuration that will allow users to set
```

Comment:
We shouldn't name a *flag* or *field* here.

`This proposal is introducing a way to allow users to set, per-sidecar, the nature of termination of a sidecar inside a task` should be enough. A flag or field is implementation detail, that can be tackle below :wink: